### DISCUSSION TITLE
PassThru driver support for upcomming WiCAN Pro hardware

### DISCUSSION THREAD

**Discussion by fowi4hjte** [meatpiHQ/wican-fw#252] [General]
The WiCAN Pro supports this OBD-II protocols with a dedicated chip:

    ISO 15765-4 CAN J1962
    SAE J1939 CAN J1962
    ISO 11898 raw CAN J1962
    Medium Speed CAN MS-CAN J1962
    GM high speed J1962
    Single Wire CAN GMLAN J1962
    SAE J1850-PWM
    SAE J1850-VPW
    ISO 9141-2
    ISO 14230-4 slow/fast

Compatible with multiple instruction sets:

    ELM327 instruction v2.3
    ELM329 instruction v2.2
    STN instruction
    VT instruction

This in combination with a opensource driver for all possible OS combinations could be beneficial to a endless amount of tools that exist out there.
For example people are ready to pay in comparison to a WiCAN Pro a much money for a original tactrix openport 2.0 hardware. But OpenPort 2.0 is compared to WiCAN Pro not that great when its about modern cars and modern protocols.
Many people buy the OpenPort 2.0 just because of its drivers and support that is been added for it over the years. These days PassThru is the most common thing that is used for diagnostics. When adding support for many tools out there the WiCAN Pro could be the next big thing everyone would be pointed to get when some service on the car have to be done.

Let me name a practical example: I have a modern BLE OBD adapter. Sadly there is a older app where the developer do not bother to build BLE support into it. The App is using STN instructions for CAN filtering. WiCAN Pro could act as a BT instead of BLE adapter and i could use this outdated app. And an other benefit would be then to have hopefully the ability to log every single command of the App to reverse its functions and build them into some open source software.
The possible use cases of WiCAN pro are really endless and it would be great to make use of them to make more things open source by giving the people the ability to debug everything.

**Comment by meatpiHQ** [comment]
@fowi4hjte Just a quick note here, the WiCAN PRO cannot act as BT. It is based on ESP32-S3 which only support BLE.

Support PassThru would be awesome, I get asked about that all the time. And I never found an easy way to do it. I could write the drivers, but my biggest issue would be testing, I have no idea what would the test setup look like.

**Reply by fowi4hjte** [reply]
> I have no idea what would the test setup look like.

I for example know a good way to test that on my own. I am running Mitsubishi MUT-3 software with either closed source PassThru or opensource PassThru driver.
The closed source PassThru driver is this one: https://www.tactrix.com/index.php?option=com_content&view=category&layout=blog&id=38&Itemid=61
The opensource driver is this here: https://github.com/kolyandex/Lexia_J2534

Yes, this is on different hardware but, i know how the working functionality is working on. For example when i use a third, broken closed source PassThru software solution, only some parts of PassThru are working and i cant access with the OEM MUT-3 software some parts of the car. MUT-3 is then crashing on those tests that are working fine with better PassThru implementations.

If i would buy WiCAN PRO i could compare it to all the other solutions i already have here.

**Comment by fowi4hjte** [comment]
About a year passed since i asked the first time about that. Have anything changed since then?
